> Yes, you mentioned in passing that it's an important component of the library. At the same time, in the limited scope of this PR (just to update the GH workflows) I thought that given that it's not a mandatory dependency and the codebase currently has support to disable use of that library (through the compiler definition `WITH_LIBXML2`), the QA runs should try and cover that configuration.

With regards to dependencies, I think the section in [README.md](https://github.com/owasp-modsecurity/ModSecurity/blob/v3/master/README.md#dependencies) could be updated to be more accurate. The line that mentions YAJL, libpcre & libXML2 starts by stating that these are mandatory dependencies but clarifies that two out of the three are not yet so. I think either pcre or pcre2 are mandatory to build the library, while as we discussed libXML2 seems optional (at least for the time). I think that sentence would be more accurate and read better like this:

- Other dependencies include YAJL, as ModSecurity uses JSON for producing logs and its testing framework, libpcre  for processing regular expressions in SecRules, and libXML2 (not yet mandatory) which is used for parsing XML requests.

One other minor issue is that the last sentence implies that libinjection is not mandatory, and that if the library is missing you just won't get the associated operator, but something I found out a few times when I forgot to init & update submodules on my Unix builds is that configure won't complete without it, which actually makes it  mandatory. :-)

```
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
configure: error: 

  libInjection was not found within ModSecurity source directory.

  libInjection code is available as part of ModSecurity source code in a format
  of a git-submodule. git-submodule allow us to specify the correct version of
  libInjection and still uses the libInjection repository to download it.

  You can download libInjection using git:

     $ git submodule init
     $ git submodule update
```

(edit) I ended up creating another quick PR (#3145) to move this discussion there and update README, but I think it'd be better if I create a PR to actually update configure and the codebase to enforce the mandatory dependencies (such as YAJL, libXML2 and probably PCRE2 -as the original pcre has been end of life since 2021-).

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -84,6 +84,7 @@ jobs:
           - {label: "full",       opt: "" }
           - {label: "wo lmdb",    opt: "-DWITHOUT_LMDB=ON" }
           - {label: "wo lua",     opt: "-DWITHOUT_LUA=ON" }
+          - {label: "wo libxml2", opt: "-WITHOUT_LIBXML2=ON" }
           - {label: "wo maxmind", opt: "-DWITHOUT_MAXMIND=ON" }
           - {label: "wo curl",    opt: "-DWITHOUT_CURL=ON" }
     steps:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -25,7 +25,7 @@ jobs:
       - name: Setup Dependencies
         run: |
           sudo apt-get update -y -qq
-          sudo apt-get install -y libfuzzy-dev libyajl-dev libgeoip-dev liblua5.2-dev liblmdb-dev cppcheck libmaxminddb-dev libcurl4-openssl-dev libpcre2-dev pcre2-utils
+          sudo apt-get install -y libfuzzy-dev libyajl-dev libgeoip-dev liblua5.2-dev liblmdb-dev libmaxminddb-dev libcurl4-openssl-dev libpcre2-dev pcre2-utils
       - uses: actions/checkout@v2
         with:
           submodules: true
@@ -38,8 +38,6 @@ jobs:
         run: make -j `nproc`
       - name: check
         run: make check
-      - name: check-static
-        run: make check-static
 
   build-macos:
     runs-on: ${{ matrix.os }}
@@ -125,3 +123,21 @@ jobs:
         working-directory: build\win32\build
         run: |
           ctest -C ${{ matrix.configuration }} --output-on-failure
+
+  cppcheck:
+    runs-on: [ubuntu-22.04]
+    steps:
+      - name: Setup Dependencies
+        run: |
+          sudo apt-get update -y -qq
+          sudo apt-get install -y cppcheck
+      - name: Get libModSecurity v3 source
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Configure libModSecurity
+        run: |
+          ./build.sh
+          ./configure
+      - name: Run cppcheck on libModSecurity
+        run: make check-static

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -26,7 +26,7 @@ jobs:
         run: |
           sudo apt-get update -y -qq
           sudo apt-get install -y libfuzzy-dev libyajl-dev libgeoip-dev liblua5.2-dev liblmdb-dev libmaxminddb-dev libcurl4-openssl-dev libpcre2-dev pcre2-utils
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
         with:
           submodules: true
       - name: build.sh
@@ -58,7 +58,7 @@ jobs:
       - name: Setup Dependencies
         run: |
           brew install autoconf automake cppcheck lmdb libyaml lua ssdeep libmaxminddb bison
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
         with:
           submodules: true
       - name: build.sh

--- a/test/regression/regression.cc
+++ b/test/regression/regression.cc
@@ -486,15 +486,12 @@ int main(int argc, char **argv) {
 #if defined(WITH_GEOIP) or defined(WITH_MAXMIND)
     resources.push_back("geoip-or-maxmind");
 #endif
-
 #if defined(WITH_MAXMIND)
     resources.push_back("maxmind");
 #endif
-
 #if defined(WITH_GEOIP)
     resources.push_back("geoip");
 #endif
-
 #ifdef WITH_CURL
     resources.push_back("curl");
 #endif
@@ -503,6 +500,9 @@ int main(int argc, char **argv) {
 #endif
 #ifdef WITH_LUA
     resources.push_back("lua");
+#endif
+#ifdef WITH_LIBXML2
+    resources.push_back("libxml2");
 #endif
 
 #ifdef NO_LOGS

--- a/test/test-cases/regression/action-ctl_request_body_processor.json
+++ b/test/test-cases/regression/action-ctl_request_body_processor.json
@@ -2,6 +2,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing CtlRequestBodyProcessor=XML (1)",
     "expected":{
       "debug_log": "Registered XML namespace href \"http://schemas.xmlsoap.org/soap/envelope/\" prefix \"soap\""
@@ -71,6 +72,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing CtlRequestBodyProcessor=XML (2)",
     "expected":{
       "debug_log": "Rule returned 0"
@@ -139,6 +141,7 @@
    {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing CtlRequestBodyProcessor=XML (3)",
     "expected":{
       "debug_log": "XML: Failed parsing document."

--- a/test/test-cases/regression/action-xmlns.json
+++ b/test/test-cases/regression/action-xmlns.json
@@ -38,6 +38,7 @@
     {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing XML request body parser (validate ok)",
     "expected":{
       "debug_log": "Target value: \"39.95\" \\(Variable: XML:\/bookstore\/book\/price\\[text\\(\\)\\]\\)"

--- a/test/test-cases/regression/config-body_limits.json
+++ b/test/test-cases/regression/config-body_limits.json
@@ -578,6 +578,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"SecRequestBodyNoFilesLimit - xml, limit exceeded",
     "client":{
       "ip":"200.249.12.31",
@@ -626,6 +627,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"SecRequestBodyNoFilesLimit - xml, limit not exceeded",
     "client":{
       "ip":"200.249.12.31",

--- a/test/test-cases/regression/config-xml_external_entity.json
+++ b/test/test-cases/regression/config-xml_external_entity.json
@@ -2,6 +2,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing SecXMLExternalEntity/XXE 1",
     "expected":{
       "debug_log": "Target value: \" jo smith\""
@@ -47,6 +48,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing SecXMLExternalEntity/XXE 2",
     "expected":{
       "debug_log": "XML: Failed to load DTD: test-cases/data/SoapEnvelope.dtd",
@@ -94,6 +96,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing SecXMLExternalEntity/XXE 3",
     "expected":{
       "debug_log": "XML Error: No declaration for element bookstore",

--- a/test/test-cases/regression/request-body-parser-xml-validade-dtd.json
+++ b/test/test-cases/regression/request-body-parser-xml-validade-dtd.json
@@ -2,6 +2,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing XML request body parser - validateDTD (validate ok)",
     "expected":{
       "debug_log": "XML: Successfully validated payload against DTD: test-cases/data/SoapEnvelope.dtd"
@@ -47,6 +48,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing XML request body parser - validateDTD (validation failed)",
     "expected":{
       "debug_log": "XML Error: No declaration for element xBody",
@@ -93,6 +95,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing XML request body parser - validateDTD (bad XML)",
     "expected":{
       "debug_log": "XML: DTD validation failed because content is not well formed",
@@ -139,6 +142,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing XML request body parser - validateDTD (bad DTD)",
     "expected":{
       "debug_log": "Failed to load DTD: test-cases/data/SoapEnvelope-bad.dtd",

--- a/test/test-cases/regression/request-body-parser-xml.json
+++ b/test/test-cases/regression/request-body-parser-xml.json
@@ -2,6 +2,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing XML request body parser (validate ok)",
     "expected":{
       "debug_log": "XML: Successfully validated payload against Schema:"
@@ -51,6 +52,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing XML request body parser (validate attribute value failed)",
     "expected":{
       "debug_log": "'badval' is not a valid value of the local atomic type",
@@ -101,6 +103,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing XML request body parser (validate failed)",
     "expected":{
       "debug_log": "This element is not expected. Expected is one of",
@@ -151,6 +154,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing XML request body parser (bad XML)",
     "expected":{
       "debug_log": "XML Error: Element '{http://schemas.xmlsoap.org/soap/envelope/}xBody'",
@@ -201,6 +205,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing XML request body parser (bad schema)",
     "expected":{
       "debug_log": "XML: Failed to load Schema: test-cases/data/SoapEnvelope-bad.xsd. XML Error: Failed to parse the XML resource 'test-cases/data/SoapEnvelope-bad.xsd",

--- a/test/test-cases/regression/variable-REQBODY_PROCESSOR_ERROR.json
+++ b/test/test-cases/regression/variable-REQBODY_PROCESSOR_ERROR.json
@@ -2,6 +2,7 @@
   {  
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing Variables :: REQBODY_PROCESSOR_ERROR_MSG (1/2)",
     "client":{  
       "ip":"200.249.12.31",

--- a/test/test-cases/regression/variable-XML.json
+++ b/test/test-cases/regression/variable-XML.json
@@ -2,6 +2,7 @@
   {
     "enabled":1,
     "version_min":300000,
+    "resource":"libxml2",
     "title":"Testing XPath expression with equals sign",
     "expected":{
       "http_code": 403

